### ISSUE TITLE
prepareAsync() lags

### ISSUE THREAD

**Issue by Dimezis** [Bilibili/ijkplayer#1507]
The following code lags for like a 0.4 - 0.7 seconds on Nexus 5.

```
player.reset();
player.setDataSource(uri)
player.prepareAsync();
```
If I remove `prepareAsync()`, the lag disappears.

I am using ExecutorService to run this task and prevent lag, but it creates additional problems of synchronization in case of long prepare.
Can you please investigate this issue or provide regular `prepare()` method to be executed synchronously?

Thank you.

**Comment by bbcallen** [comment]
AFAICT, if you remove prepareAsync(), the player won't start, at all. How did the lag disappear?

The prepare action is asynchronous in native, I would prefer to wrap a synchronous method in java.

**Comment by Dimezis** [comment]
> AFAICT, if you remove prepareAsync(), the player won't start, at all. How did the lag disappear?

Yes, of course. I removed `prepareAsync()` just for testing. After I moved it to ExecutorService task, the lag disappeared.
 
> I would prefer to wrap a synchronous method in java.

That would be great.

**Comment by bbcallen** [comment]
I did't mean it's on my plan, I mean it COULD be implemented in java.
